packages/@webex/internal-plugin-encryption/test/unit/spec/kms-errors.js

```diff
@@ -0,0 +1,51 @@
+import {assert} from '@webex/test-helper-chai';
+import {handleKmsKeyRevokedEncryptionFailure} from '../../../src/kms-errors'
+import sinon from 'sinon';
+
+import {KMS_KEY_REVOKE_FAILURE, KMS_KEY_REVOKE_ERROR_STATUS, KMS_KEY_REVOKE_ERROR_CODES} from '../../../src/constants'
+
+describe('handleKmsKeyRevokedEncryptionFailure', () => {
+
+    it('triggers `event:kms:key:revoke:encryption:failure` event when correct error codes are detected', () => {
+        const webex = {
+            internal: {
+                encryption: {
+                trigger: sinon.spy()
+              },
+            },
+          }
+
+        const item = {
+            status: KMS_KEY_REVOKE_ERROR_STATUS,
+            body: {
+                errorCode: KMS_KEY_REVOKE_ERROR_CODES[0],
+            }
+        }
+
+        handleKmsKeyRevokedEncryptionFailure(item, webex);
+        
+        assert.calledOnce(webex.internal.encryption.trigger);
+        assert.calledWithExactly(webex.internal.encryption.trigger, KMS_KEY_REVOKE_FAILURE);
+    });
+
+    it('does not trigger `event:kms:key:revoke:encryption:failure` event when other error codes are detected', () => {
+        const webex = {
+            internal: {
+                encryption: {
+                trigger: sinon.spy()
+              },
+            },
+          }
+
+        const item2 = {
+            status: KMS_KEY_REVOKE_ERROR_STATUS,
+            body: {
+                errorCode: 403009,
+            }
+        }
+
+        handleKmsKeyRevokedEncryptionFailure(item2, webex);
+         
+         assert.notCalled(webex.internal.encryption.trigger);
+       });
+});
```

Comment:
It might be worth considering actually having a listener set up in another test that *actually* gets the event that the SDK triggered, i.e.,

webex.internal.encryption.on('event', () => {})

runAllTicks()

expect(funcToBeCalled)

but this might be overkill and i won't reject it based on this